SwiftReflector/Inventory/ModuleInventory.cs

```diff
@@ -111,6 +114,23 @@ public static ModuleInventory FromFile (string pathToDynamicLibrary, ErrorHandli
 			}
 		}
 
+		public static async Task<ModuleInventory> FromFilesAsync (IEnumerable<string> pathsToLibraryFiles, ErrorHandling errors)
+		{
+			var streams = pathsToLibraryFiles.Select (path => new FileStream (path, FileMode.Open, FileAccess.Read, FileShare.Read)).ToList ();
+			if (streams.Count == 0)
+				return null;
+			var inventory = new ModuleInventory ();
+			try {
+				var tasks = streams.Select (stm => FromStreamIntoAsync (stm, inventory, errors, stm.Name));
+				await Task.WhenAll (tasks);
+			} finally {
+				foreach (var stm in streams) {
+					stm.Dispose ();
+				}
```

Comment:
The Reactive Extensions (https://github.com/dotnet/reactive)  provides a nice way to deal with a collection of IDisposables: https://learn.microsoft.com/en-us/previous-versions/dotnet/reactive-extensions/hh229918(v=vs.103)

I don't know how you feel about adding it as a dependency "System.Reactive" but will probably help.